Doc/deprecations/pending-removal-in-3.15.rst

```diff
@@ -51,6 +51,11 @@ Pending removal in Python 3.15
     This function is only useful for Jython support, has a confusing API,
     and is largely untested.
 
+* :mod:`sysconfig`:
+
+  * The ``check_home`` argument of :func:`sysconfig.is_python_build` has been
+    deprecated since Python 3.12.
```

Comment:
Let's also remove the old deprecation from "Pending removal in future versions":

https://github.com/python/cpython/blob/56e190068177855266f32a7efa329d145b279f94/Doc/deprecations/pending-removal-in-future.rst#L114-L115

I'll open a new PR to remove it from `main`, and include it in the backports of this PR.